### DISCUSSION TITLE
Is there a way to modify the user's message?

### DISCUSSION THREAD

**Discussion by hexsix** [telegraf/telegraf#1683] [Q&A] [answered]
I want to modify some link messages of the user (not bot). For Example, `https://i.pximg.net/xxx` to `https://pixiv.net/xxx`, etc

Here's the code I tried but both not work.

```typescript
bot.on('text', (ctx) => {
    ctx.telegram.editMessageText(
        ctx.message.chat.id,
        ctx.message.message_id,
        undefined,
        'modified',
    )
  });
```

```typescript
bot.on('text', (ctx) => {
    ctx.editMessageText(
        'modified',
    )
  });
```

Any ideas there?

**Comment by hypernova7** [comment, marked as answer]
That isn't possible with the bot API, you may need to try your luck using MTProto

**Reply by hexsix** [reply]
Alright, thank you for your reply.
